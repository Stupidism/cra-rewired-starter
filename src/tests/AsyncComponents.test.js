import React from 'react';
import { shallow } from 'enzyme';
import _ from 'lodash';

import AsyncHomePage from 'routes/HomePage';
import AsyncNotFoundPage from 'routes/NotFoundPage';

const AsyncComponents = {
  AsyncHomePage,
  AsyncNotFoundPage,
};

describe('<AsyncComponent />', () => {
  _.forEach(AsyncComponents, (AsyncComponent, name) => {
    test(`<${name} /> should render a Loader first and then itself`, async () => {
      expect(AsyncComponent.name).toEqual('LoadableComponent');
      const wrapper = shallow(<AsyncComponent />);
      expect(wrapper.dive()).toMatchSnapshot();
      await AsyncComponent.preload();
      wrapper.update();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
