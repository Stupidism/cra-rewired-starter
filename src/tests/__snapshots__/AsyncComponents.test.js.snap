// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsyncComponent /> <AsyncHomePage /> should render a Loader first and then itself 1`] = `
<Loading
  className="Loading"
  color="#262729"
  delay={0}
  height={64}
  type="spokes"
  width={64}
/>
`;

exports[`<AsyncComponent /> <AsyncHomePage /> should render a Loader first and then itself 2`] = `<Component />`;

exports[`<AsyncComponent /> <AsyncNotFoundPage /> should render a Loader first and then itself 1`] = `
<Loading
  className="Loading"
  color="#262729"
  delay={0}
  height={64}
  type="spokes"
  width={64}
/>
`;

exports[`<AsyncComponent /> <AsyncNotFoundPage /> should render a Loader first and then itself 2`] = `<Component />`;
